snapcraft/internal/meta/plugs.py

```diff
@@ -54,15 +54,16 @@ def validate(self) -> None:
     def from_dict(cls, *, plug_dict: Dict[str, Any], plug_name: str) -> "Plug":
         """Create plug from dictionary."""
 
-        interface = plug_dict.get("interface", None)
-        if interface is not None:
-            plug_class = PLUG_MAPPINGS.get(interface, None)
-            if plug_class is not None:
-                return plug_class.from_dict(plug_dict=plug_dict, plug_name=plug_name)
+        plug_dict = {} if plug_dict is None else plug_dict
```

Comment:
Just my 0.02PLN, the assignment, especially using the rather verbose python conditional syntax is probably not saving much over the very explicit:

```
if plug_dict is None:
  plug_dict = {}
```

Reply:
I've got some other related changes in queue, I can update in that PR.

Reply:
Relatedly, the plug_dict needs to be labelled Optional if allowing None, but I'll be following up with that as well.

Reply:
> Just my 0.02PLN

For me both assignments are ok. in the past I had always saved lines.

Reply:
> Relatedly, the plug_dict needs to be labelled Optional if allowing None, but I'll be following up with that as well.

@cjp256 
In the past years i only used python 2.7 on a Solaris 10 machine. So for me, you are the master with "labels" ;-)